docs/src/development/proposals/MEP15/index.md

```diff
@@ -0,0 +1,168 @@
+
+Over the past few years, it has become clear that potential users are always interested in using more and different types of hardware, either because they want to reuse their existing hardware or because their companies are tied to specific vendors.
+It would be a great improvement for them to have broader support for hardware in general, similar to what is promised by projects like OpenStack Ironic.
+
+We have found that vendor support is hard to implement and even harder to test and maintain.
+We have some really sophisticated parts in our code base, reaching down to patching BIOS XMLs for individual BIOS versions of specific motherboards.
+It is almost impossible to touch these pieces of code again because it could break the implementation for specific hardware.
+
+So with this MEP, we want to evaluate ways to improve our code to make it easier to add new vendors, increase the reliability of the implementation, and provide broader hardware support more quickly.
+
+While we continue to have a list of vendors and models for which we verified our integration works, we will also be able to say that we have general support for a number of drivers, starting with IPMI, Redfish, iDrac and Unmanaged (e.g. for developer usage in the mini-lab).
+
+Vendors that implement these driver APIs properly may work right out of the box through our default implementation for a given driver. Through a CLI operators can quickly figure out if the existing implementation is sufficient or not. If a vendor requires specific modifications of the default implementation a dedicated vendor-overwrite can be implemented in go-hal (which will be required for Supermicro for sure).
+
+## Shortcomings of the current implementation
+
+- Every new vendor has to be individually whitelisted in go-hal, a new board of a Supermicro server potentially requires a pull request.
+- The current interface implements functions using different underlying drivers whenever it fits, so it's not obvious if IPMI or Redfish is used, sometimes information from different protocols differ.
+- There are almost no unit tests and no automated integration tests (despite indirect integration testing through our release integration, which is creating and deleting machines).
+- The CLI is not implementing the entire API interface. It is implemented only roughly. To use it for testing, it requires ad hoc code changes and recompilation.
+- There is no possibility for an operator to provide the user / password that a server was shipped with in order to intialize it with these credentials. Such that the metal-hammer must be booted first before it can be managed through BMC. This also implies that the implemtation relies on inband to work.
+
+## Bundling Functionality in the metal-bmc
+
+In order to minimize the BMC interface and spreaded library usage, we should try to bundle as much of the implementation as possible in a single microservice. This microservice should have a proto / gRPC API for access.
```

Comment:
It is actually not spreaded, but only used in the metal-bmc (outband) and metal-hammer (inband).

Reply:
Removed this part of the sentence.